sdk/data_kusto/src/operations/query.rs

```diff
@@ -0,0 +1,174 @@
+use crate::client::KustoClient;
+use crate::request_options::*;
+use azure_core::prelude::*;
+use azure_core::setters;
+use azure_core::{
+    collect_pinned_stream,
+    headers::{add_mandatory_header2, add_optional_header2},
+    Response as HttpResponse,
+};
+use futures::future::BoxFuture;
+
+/// A future of a delete file response
+type ExecuteQuery = BoxFuture<'static, crate::error::Result<KustoResponseDataSetV2>>;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct QueryBody {
+    db: String,
+    csl: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct ExecuteQueryBuilder {
+    client: KustoClient,
+    database: String,
+    query: String,
+    client_request_id: Option<ClientRequestId>,
+    context: Context,
+}
+
+impl ExecuteQueryBuilder {
+    pub(crate) fn new(
+        client: KustoClient,
+        database: String,
+        query: String,
+        context: Context,
+    ) -> Self {
+        Self {
+            client,
+            database,
+            query,
+            client_request_id: None,
+            context,
+        }
+    }
+
+    setters! {
+        client_request_id: ClientRequestId => Some(client_request_id),
+        query: String => query,
+        database: String => database,
+        context: Context => context,
+    }
+
+    pub fn into_future(self) -> ExecuteQuery {
+        let this = self.clone();
+        let ctx = self.context.clone();
+
+        Box::pin(async move {
+            let url = this.client.query_url();
+            let mut request = this.client.prepare_request(url, http::Method::POST);
+
+            add_mandatory_header2(
+                &ContentType::new("application/json; charset=utf-8"),
+                &mut request,
+            )
+            .unwrap();
+            add_mandatory_header2(&Accept::new("application/json"), &mut request).unwrap();
+            add_mandatory_header2(&AcceptEncoding::new("gzip,deflate"), &mut request).unwrap();
```

Comment:
This header causes the response to get back as gzip, which the azure core library doesn't seem to decypher, so I get an error unless I remove it. Is it the same experience for you?

Reply:
hmm.. when i run it locally, it actually seems to be required, strange. One way to compare would be to record a query using the mock transport framework and compare how the same response behaves on different machines. Do you maybe have a test cluster to run queries against that we can record? I could then replay the test and see if it fails or passes?

Reply:
turns out, now it's the same experience for me :D. SO removed the header for now.

Reply:
this was related to the core crate not including the `gzip` feature of `reqwest`, added it since it as an optional feature so we can use gzip in this crate.